component/src/main/java/org/wso2/extension/siddhi/gpl/execution/pmml/PmmlModelProcessor.java

```diff
@@ -303,31 +261,42 @@ private void populateFeatureAttributeMapping() throws Exception {
     }
 
     /**
-     * TODO : move to a Util class (PmmlUtil)
-     * Unmarshal the definition and get an executable pmml model.
+     * Generate the output attribute list.
      *
-     * @return pmml model
+     * @return List
      */
-    private PMML unmarshal(String pmmlDefinition) {
-
-        try {
-            File pmmlFile = new File(pmmlDefinition);
-            InputSource pmmlSource;
-            Source source;
-            // if the given is a file path, read the pmml definition from the file
-            if (pmmlFile.isFile() && pmmlFile.canRead()) {
-                pmmlSource = new InputSource(new FileInputStream(pmmlFile));
-            } else {
-                // else, read from the given definition
-                pmmlSource = new InputSource(new StringReader(pmmlDefinition));
+    private List<Attribute> generateOutputAttributes() {
+        List<Attribute> outputAttributes = new ArrayList<>();
+        for (Map.Entry<FieldName, org.dmg.pmml.DataType> entry : outputFields.entrySet()) {
+            FieldName fieldName = entry.getKey();
+            org.dmg.pmml.DataType dataType = entry.getValue();
+            if (dataType == null) {
+                dataType = org.dmg.pmml.DataType.STRING;
             }
-            source = ImportFilter.apply(pmmlSource);
-            return JAXBUtil.unmarshalPMML(source);
-        } catch (SAXException | JAXBException | FileNotFoundException e) {
-            logger.error("Failed to unmarshal the pmml definition: " + e.getMessage());
-            throw new SiddhiAppCreationException("Failed to unmarshal the pmml definition: " + pmmlDefinition + ". "
-                    + e.getMessage(), e);
+            outputAttributes.add(new Attribute(fieldName.getValue(), mapOutputAttributes(dataType)));
+        }
+        return outputAttributes;
+    }
+
+    /**
+     * Map the model output fields to Siddhi Attributes.
+     *
+     * @return Attribute.Type
+     */
+    private Attribute.Type mapOutputAttributes(org.dmg.pmml.DataType dataType) {
+        Attribute.Type type = null;
+        if (dataType.equals(org.dmg.pmml.DataType.DOUBLE)) {
+            type = Attribute.Type.DOUBLE;
+        } else if (dataType.equals(org.dmg.pmml.DataType.FLOAT)) {
+            type = Attribute.Type.FLOAT;
+        } else if (dataType.equals(org.dmg.pmml.DataType.INTEGER)) {
+            type = Attribute.Type.INT;
+        } else if (dataType.equals(org.dmg.pmml.DataType.BOOLEAN)) {
+            type = Attribute.Type.BOOL;
+        } else if (dataType.equals(org.dmg.pmml.DataType.STRING)) {
+            type = Attribute.Type.STRING;
```

Comment:
Add a else case as well and throw a exception with corresponding class

Reply:
Method updated in commit [#82955b8dc64c8ad2fb9bb41b8c0be55ecfb67b3a](https://github.com/wso2-extensions/siddhi-gpl-execution-pmml/pull/32/commits/82955b8dc64c8ad2fb9bb41b8c0be55ecfb67b3a)